co-simulation/fed/mosaic-infrastructure/pom.xml

```diff
@@ -25,6 +25,13 @@
       <artifactId>mosaic-objects</artifactId>
       <version>${mosaic.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.eclipse.mosaic</groupId>
+      <artifactId>mosaic-objects</artifactId>
+      <version>${mosaic.version}</version>
+      <type>test-jar</type>
+      <scope>test</scope>
+    </dependency>
     <dependency>
         <groupId>gov.dot.fhwa.saxton</groupId>
         <artifactId>mosaic-carma-utils</artifactId>
```

Comment:
Can we make this mosaic-carma-utils artifact group id consistent as the other artifacts in the lib? Eg: mosaic-objects is in group org.eclipse.mosaic, but the mosaic-carma-utils is in group gov.dot.fhwa.saxton?

Reply:
No mosiac-carma-utils is a jar we created while mosiac-objects is a jar created by eclipse mosiac. This may mean that I need to move my detection objects into a mosiac-carma-objects library.

Reply:
I do not think we need to move the detection object into library. This is just group ID changes to move `mosaic-carma-utils` into `org.eclipse.mosaic` group. And replace the namespace `gov.dot.fhwa.saxton` in the `mosaic-carma-utils` to `org.eclipse.mosaic`

Reply:
This is the group id I refer to: https://github.com/usdot-fhwa-stol/cdasim/blob/35c45a53d499c7ba8901177ffd73f5b6a0d32fab/co-simulation/lib/mosaic-carma-utils/pom.xml#L13.
The package is here: https://github.com/usdot-fhwa-stol/cdasim/blob/35c45a53d499c7ba8901177ffd73f5b6a0d32fab/co-simulation/lib/mosaic-carma-utils/src/main/java/gov/dot/fhwa/saxton/CarmaV2xMessage.java#L17

https://github.com/usdot-fhwa-stol/cdasim/blob/35c45a53d499c7ba8901177ffd73f5b6a0d32fab/co-simulation/lib/mosaic-carma-utils/src/main/java/gov/dot/fhwa/saxton/CarmaV2xMessageReceiver.java#L17

Reply:
I am aware of the group id. That is mean to indicate the organization responsible for creating the code. Mosiac is an Eclipse managed open source simulation framework we developed on top of.  The packages we developed are developed under the package structure gov.dot.fhwa.saxton to indicate they are stol packages while the mosiac packages are developed under the org.eclipse.mosiac.lib package structure

Reply:
https://github.com/eclipse/mosaic

Reply:
I see. It is a bit confusing, if it is a clone a another repos, not sure why we made a full copy here instead of just using a clone reference to that repos. This mosaic-carma-utils package is located under this lib but has a completely out of context package name `gov.dot.fhwa.saxton` from my point of view. Initially, I thought this mosaic-carma-utils is from another repos because of the package name `gov.dot.fhwa.saxton`.

Reply:
I agree that it would have been better to include mosiac as a maven dependency but completely disagree with your statement about package structure being confusing. I think doing this kind of package structure is standard practice for extending Java frameworks/libraries. The package structure gives insight not only to what functionality the code inside provides but whether it is part of the core framework or not and what organization contributed this extension.

Reply:
@kjrush if you have input on whether this package structure is confusing and also whether I should avoid adding to eclipse packages like I have in this PR and instead put my code in a fhwa library like carma utils one please let me know

Reply:
For context, the rationale behind cloning the repo rather than simply linking the maven artifacts was simply that we had full intent to be modifying the core simulator packages in order to add CARLA as a whole new class of simulator. We don't truly need every package but I feel like a patchwork solution is likely more confusing than just maintaining our own fork.

Overall though, I agree with your assessment on the group name. The group name is intended to indicate maintainer as you said, such that you could have multiple versions of a package maintained by different groups with possibly different implementations or features. That does make me wonder if truly the correct solution here is to actually migrate all the packages to gov.dot.fhwa.stol, since we're maintaining this fork, but I'm not sure that's a question to be resolved here.